Automattic/wp-calypso
# Launchpad: Removes the Navigator request calls

Related to paYKcK-3CA-p2

## Proposed Changes

* Since we decided to pause the development of the Launchpad Navigator, we decided to reduce the number of API calls that we're making for the feature. To do that, we're doing a few things in this PR:
 - We're moving most `setActiveChecklist` calls behind the existing `launchpad/navigator` feature flag.
 - We're removing the logic to fetch and update the active checklist from the Customer Home launchpad component
    * If we decide to work on the feature in future, we can add the relevant logic back in.

## Testing Instructions



* Run this branch locally or via Calypso.live as per the instructions in [this comment](https://github.com/Automattic/wp-calypso/pull/83835#issuecomment-1791762699)
* For one or more of the following flows, work through the signup flow until you get to the fullscreen launchpad
  - `/setup/build` - create a new site via `/start` and choose the build intent ("Promote myself or business")
  - `/setup/design-first` - navigate to `/setup/design-first?ref=calypshowcase`
  - `/setup/free` - create a new site via `/setup/free`
  - `/setup/link-in-bio` - create a new site via `/setup/link-in-bio`
  - `/setup/link-in-bio-tld` - create a new site via `/setup/link-in-bio-tld`, and make sure to click on "Decide later" on the domains step
  - `/setup/newsletter` - create a new site via `/setup/newsletter`
  - `/setup/start-writing` - this requires you to start the flow with a test user that doesn't have any sites
  - `/setup/videopress` - create a new site via `/setup/videopress`
* When you get to the fullscreen launchpad, open your browser network tools, and click on the "Skip for now" option in the fullscreen launchpad
* Verify that we did not trigger a call to `PUT /wpcom/v2/sites/:siteSlug/launchpad/navigator`
* You should then be sent to Customer Home showing the same set of tasks
* Verify that in Customer Home you do not see a `GET /wpcom/v2/sites/:siteSlug/launchpad/navigator` API call.
* Set up a site from one of the flows above and work through the flow until you get to the fullscreen launchpad
* Add `&flags=launchpad/navigator` to the URL when you're in the fullscreen launchpad 
* Once the page has fully reloaded, click on the skip link
* Verify that we make the `PUT /wpcom/v2/sites/:siteSlug/launchpad/navigator` request when the feature flag is enabled.
* When you get to Customer Home, add `?flags=launchpad/navigator` to the URL
* After the page has reloaded, click on a task in the last list
* Verify that we make a `PUT /wpcom/v2/sites/:siteSlug/launchpad/navigator` request in response to the click

## Pre-merge Checklist



- [N/A] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
